packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -255,7 +285,45 @@ export class Pipeline extends cdk.Construct implements cpapi.IPipeline {
 
   // ignore unused private method (it's actually used in Stage)
   // @ts-ignore
-  private _generateOutputArtifactName(stage: cpapi.IStage, action: cpapi.Action): string {
+  private _attachAction(stage: Stage, action: actions.Action): void {
+    // handle cross-region Actions here
+    if (action.region) {
+      // get the region the Pipeline itself is in
+      const pipelineStack = cdk.Stack.find(this);
+      const pipelineRegion = pipelineStack.requireRegion(
+        "You need to specify an explicit region when using CodePipeline's cross-region support");
+      const pipelineAccount = pipelineStack.requireAccountId(
+        "You need to specify an explicit account when using CodePipeline's cross-region support");
+      const app = pipelineStack.parent === undefined ? undefined : pipelineStack.parent as cdk.App;
```

Comment:
Just to be safe, I would also verify that the construct name is "App". For example, I wonder if at some point we will allow people to define stacks as children of other stacks (i.e. nested stacks) and then this assumption will break.

```ts
pipelineStack.constructor.name === "App"
```

Reply:
I'm not exactly sure what you mean, but I've extracted a method in `Stack` called `parentApp`, the logic now lives there. Hope that's safe enough (also, if we allowed `Stacks` as parents, we would need to to change the API of `CrossRegionScaffoldStack` to match, which would change this code.